File: App/App6/servidor/src/api/palabra_clave_routes.py
from controllers.palabra_clave_controller import PalabraClave_controller

def palabras_clave_register_routes(app):
    """Define las rutas registradas para las palabras clave"""
    # Rutas de palabras clave
    app.route("/palabras_clave/bot/<bot_id>", methods=['GET'])(PalabraClave_controller.get_palabras_clave_by_bot)
    app.route("/palabras_clave", methods=['POST'])(PalabraClave_controller.add_palabra_clave)
    app.route("/palabras_clave/<palabra_clave_id>", methods=['GET'])(PalabraClave_controller.get_palabra_clave)
    app.route("/palabras_clave/<palabra_clave_id>", methods=['PUT'])(PalabraClave_controller.update_palabra_clave)
    app.route("/palabras_clave/<palabra_clave_id>", methods=['DELETE'])(PalabraClave_controller.delete_palabra_clave)

File: App/App6/Cliente/src/api/PalabraClaveRoutes.js
import { api_ip } from "../../Config";
const API = api_ip + '/palabras_clave'; // Actualiza la ruta base para las palabras clave

export const deletePalabraClave = async (palabraClaveId) => {
  await fetch(`${API}/${palabraClaveId}`, {
    method: "DELETE",
  });
};

export const getPalabrasClaveByBot = async (botId) => {
  const res = await fetch(`${API}/bot/${botId}`, {
    method: "GET",
  });

  return await res.json();
};

export const savePalabraClave = async (newPalabraClave) => {
  const res = await fetch(`${API}`, {
    method: "POST",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(newPalabraClave),
  });
  return await res.json();
};

export const getPalabraClave = async (palabraClaveId) => {
  const res = await fetch(`${API}/${palabraClaveId}`);
  const palabraClave = await res.json();
  return palabraClave;
};

export const updatePalabraClave = async (palabraClaveId, updatedPalabraClave) => {
  const res = await fetch(`${API}/${palabraClaveId}`, {
    method: "PUT",
    headers: {
      Accept: "application/json",
      "Content-Type": "application/json",
    },
    body: JSON.stringify(updatedPalabraClave),
  });
  return res;
};
